pkg/controller/node/status.go

```diff
@@ -177,6 +177,7 @@ func isNodeReady(node *corev1.Node) bool {
 	return true
 }
 
+// TODO(runcom): drop currentConfig arg
```

Comment:
I think actually this argument is acting as "gather nodes for pool", since it effectively filters them to just the pool's config.

Now is `getCandidateMachines()` trying to filter to just nodes owned by a pool?  Um...maybe?  I find all of these undocumented functions confusing...

Reply:
at the time we're here, we already have the subset of nodes for our pool though, that's why I don't think that arg is necessary...

Reply:
The thing with this is that a pool can only have a rendered config at a time it's syncing. If we allow 2 nodes to have different configs and progress them indipendently, we'll end up with the issue we're seeing. This shouldn't gate everything by saying: alrigth, everything must be idle in general in order to progress.